android-kotlin/app/src/main/java/com/mongodb/app/ItemsActivity.kt

```diff
@@ -1,39 +1,154 @@
 package com.mongodb.app
 
+import android.app.AlertDialog
 import android.content.Intent
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
 import android.util.Log
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
 import android.widget.Button
+import android.widget.EditText
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.Toolbar
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.floatingactionbutton.FloatingActionButton
 import io.realm.Realm
+import io.realm.kotlin.where
+import io.realm.mongodb.User
+import io.realm.mongodb.sync.SyncConfiguration
+
 
 class ItemsActivity : AppCompatActivity() {
-    private lateinit var logoutButton: Button
+    private lateinit var fab: FloatingActionButton
+    private lateinit var adapter: ItemAdapter
+    private lateinit var recyclerView: RecyclerView
     private var userRealm: Realm? = null
+
+    override fun onStart() {
+        super.onStart()
+        val user = realmApp.currentUser()
+        if (user == null) {
+            // if no user is currently logged in, start the login activity so the user can authenticate
+            startActivity(Intent(this, LoginActivity::class.java))
+        }
+        else {
+            // get the partition value and name of the project we are currently viewing
+            val partition = realmApp.currentUser().toString()
```

Comment:
I'm not 100% familiar with the Java/Kotlin SDK so I'm not sure, but this doesn't quite look correct. I think the partition value should be the current user's ID. This looks like it might serialize the actual user object. (Which would technically "work" until you try to see your tasks on the iOS client, which does use the user ID). Can you confirm?